src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
-class nodeA1,nodeA3,nodeA6,nodeA7 steel1
+classDef dafault_node1 fill:#ECECFF
+class tokenize,optimize,generate steel1
+class Source,Executable daisy1
+class create,execute dafault_node1
 ```
+As one can see from the picture, overall plugin pipeline consists of two major blocks: the first block applies transformations to `nGraph model` while the second one works with the internal plugin graph representation. Since `Snippets` is a backend-independent framework, it can't work with the plugin graph or plugin-specific `Ops` directly, so the tokenization is performed immediately before plugin-specific operations are introduced into the graph (`Conversion to Plugin opset`). `Tokenizer` replaces parts of the `nGraph model` that can be executed by `Snippets` with `ov::op::Subgraph` nGraph nodes. Each of the nodes stores a piece of the initial `nGraph model` that was replaced by the node. This piece is stored as an nGraph model itself, which we refer to  as `nGraph IR` to distinguish from the original `nGraph model`. Note that sometimes the exact type of `IR` is not important in our discussion. In such cases, we will refer to the `IR` (`nGraph` or `Linear`) as `body function`, or simply `body`. 
 
-If Load followed by Broadcast is detected, then this pair is replaced by a single BroadcastLoad instruction:
+When the plugin finalizes all `nGraph model` transformations, the model is converted to an internal plugin graph representation. At this point `ov::op::Subgraph` is converted to `ov::intel_cpu::node::Snippet` which still retains the `nGraph IR`. This IR is then optimized and an executable `Kernel` is produced during the `CreatePrimitive()` stage. Finally, multiple copies of the produced kernel executed in parallel during the `Execute` stage.
 
+This brief description finalizes our discussion of Snippets architecture.
+To summarize, `Snippets` workflow consists of three major blocks: `Tokenizer`, `Optimizer` and `Generator`. Let's discuss each of them in detail below.
+
+### Tokenizer
+
+`Tokenizer` is run on an `nGraph model` and its main purpose is to identify subgraphs that are suitable for code generation. These subgraphs are then replaced with the `ov::op::Subgraph` node. This stage is called tokenization because the `Tokenizer` employs a greedy  algorithm similar to the ones used for parsing input stream of characters into tokens. One of the distinctive features of this algorithm is its flexibility, so it can seamlessly handle arbitrary operations' patterns. This differentiates the tokenization from the pattern matching approach used in traditional and MLIR-based compilers. Pattern matching can indeed process only a limited set of predefined operations' configurations, so the relations between the operations are fixed in this case. Thus, the tokenization's flexibility becomes a significant advantage when the number of new ML topologies increases rapidly, so it becomes more and more expensive to support and extend a large set of patterns.
```

Comment:
Should we add some words about `Tokenization` in general? `Tokenization` has the pipeline of tokenization: `Enumeration` (to enumerate nodes for cycle dependency verification), `MHATokenization` (the pass which tokenizes MHA patterns), `CommonTokenizations` (...) and `CommonOptimizations` (some optimizations before conversion to snippet dialect)

What do you think? I'm not sure that we should explain how `MHATokenization` works but maybe several words and maybe link to article with this pattern and mermaid with skeleton of pattern.